src/libraries/System.Private.CoreLib/src/System/Reflection/Emit/TypeBuilderInstantiation.cs

```diff
@@ -253,7 +279,72 @@ public override bool ContainsGenericParameters
 
         [RequiresUnreferencedCode("If some of the generic arguments are annotated (either with DynamicallyAccessedMembersAttribute, or generic constraints), trimming can't validate that the requirements of those annotations are met.")]
         public override Type MakeGenericType(params Type[] inst) { throw new InvalidOperationException(SR.Format(SR.Arg_NotGenericTypeDefinition, this)); }
-        public override bool IsAssignableFrom([NotNullWhen(true)] Type? c) { throw new NotSupportedException(); }
+
+        [UnconditionalSuppressMessage("ReflectionAnalysis", "IL2075:DynamicallyAccessedMemberTypes",
+            Justification = "Interfaces used in Reflection.Emit either dynamic or accessible")]
+        public override bool IsAssignableFrom([NotNullWhen(true)] Type? c)
+        {
+            if (c == null)
+            {
+                return false;
+            }
+
+            if (this == c || IsAssignableFrom(c.BaseType))
+            {
+                return true;
+            }
+
+            if (!c.IsConstructedGenericType)
+            {
+                return false;
+            }
+
+            Type typeDefinition = c.GetGenericTypeDefinition();
+
+            if (typeDefinition.IsAssignableFrom(_genericType) &&
+                CheckArguments(c.GenericTypeArguments))
+            {
+                return true;
+            }
+
+            foreach (Type iface in typeDefinition.GetInterfaces())
+            {
+                if (iface.IsConstructedGenericType &&
+                    iface.GetGenericTypeDefinition().IsAssignableFrom(_genericType) &&
+                    CheckArguments(c.GenericTypeArguments))
+                {
+                    return true;
+                }
+            }
+
+            if (c.BaseType != null &&
+                c.BaseType.IsConstructedGenericType &&
+                c.BaseType.GetGenericTypeDefinition().IsAssignableFrom(_genericType) &&
+                CheckArguments(c.GenericTypeArguments))
+            {
+                return true;
+            }
+
+            return false;
+        }
+
+        private bool CheckArguments(Type[] genericTypeArguments)
+        {
+            if (_typeArguments.Length != genericTypeArguments.Length)
+            {
+                return false;
+            }
+
+            for (int i = 0; i < _typeArguments.Length; i++)
+            {
+                if (_typeArguments[i] != genericTypeArguments[i])
```

Comment:
Interfaces support co- and contra- variance. For example:
```
Console.WriteLine(typeof(IFace<object>).IsAssignableFrom(typeof(IFace<string>)).ToString());

interface IFace<out T>
{
}
```
This prints true, but I think the ref-emit equivalent will print false.